packages/main/src/RPA/MSGraph.py

```diff
@@ -0,0 +1,956 @@
+        name: Optional[str] = None,
+    ) -> Path:
+        """Downloads file from the share link.
+
+        The downloaded file will be saved to a local path.
+
+        :param share_url: URL of the shared file
+        :param to_path: Destination folder of the downloaded file,
+                defaults to the current directory.
+        :param name: New name for the downloaded file, with or without extension.
+
+        .. code-block: robotframework
+
+            *** Tasks ***
+            Download file
+                ${download_path}=    Download File From Share Link
+                ...    https://...
+                ...    /path/to/local/folder
+                ...    Report.pdf
+        """
+        self._require_authentication()
+
+        # O365 doesn't support getting items from shared links yet
+        base_url = self.client.protocol.service_url
+        base_url = base_url[:-1] if base_url.endswith("/") else base_url
+        encoded_url = self._encode_share_url(share_url)
+        endpoint = "/shares/{id}/driveItem"
+        direct_url = "{}{}".format(base_url, endpoint.format(id=encoded_url))
+
+        response = self.client.con.get(direct_url)
+        if not response:
+            return None
+
+        data = response.json()
+        file_instance = SharedItem(data, direct_url, self.client.con)
+        return self._download_file(file_instance, to_path, name)
+
+    @keyword
+    def upload_file_to_onedrive(
+        self,
+        file_path: str,
+        target_folder: Union[drive.Folder, str, None] = None,
+        resource: Optional[str] = None,
+        drive_id: Optional[str] = None,
+    ) -> drive.DriveItem:
+        # pylint: disable=anomalous-backslash-in-string
+        """Uploads a file to the specified OneDrive folder.
+
+        The uploaded file is returned as a DriveItem object and it has
+        additional properties that can be accessed with dot-notation, see
+        \`List Files In Onedrive Folder\` for details.
+
+        :param file_path: Path of the local file being uploaded.
+        :param target_folder: Path of the folder in OneDrive.
+        :param resource: Name of the resource if not using default.
+        :param drive_id: Drive ID if not using default.
+
+        .. code-block: robotframework
+
+            *** Tasks ***
+            Upload file
+                ${file}=    Upload File To Onedrive
+                ...    /path/to/file.txt
+                ...    /path/to/folder
+        """  # noqa: W605
+        self._require_authentication()
+        drive_instance = self._get_drive_instance(resource, drive_id)
+        folder = self._get_folder_instance(drive_instance, target_folder)
+        return folder.upload_file(item=file_path)
+
+    @keyword
+    def get_sharepoint_site(
+        self, *args: str, resource: Optional[str] = ""
+    ) -> sharepoint.Site:
+        """Returns a SharePoint site.
+
+        :param args: It accepts multiple ways of retrieving a site. See below.
```

Comment:
all the ways are using `get_site`  when keyword name is `get_sharepoint_site`

Reply:
Rewrote it as a robot framework code block